### DISCUSSION TITLE
How to configure NGramAnalyzer - MinGram and MaxGram values

### DISCUSSION THREAD

**Discussion by nthonymiller** [ravendb/ravendb#17183] [Q&A]
Is it possible to set the MinGram and MaxGram values globally on the database level and/or index level?

I've tried updating on the Index Level, but the value doesn't impact the index creation. I can see the index has accepted the configuration but the terms have not been updated.

```
Index(x => x.Search, FieldIndexing.Search);
Analyzers.Add(x => x.Search, "NGramAnalyzer");

Configuration["Indexing.Analyzers.Lucene.NGram.MaxGram"] = "20";
```

**Comment by arekpalinski** [comment]
Yes, you can set it at database / index level. I think you used invalid configuration name. You can use:

- "Indexing.Lucene.Analyzers.NGram.MaxGram" (instead of "Indexing.Analyzers.Lucene.NGram.MaxGram")
- "Indexing.Lucene.Analyzers.NGram.MinGram"

Note that update of those setting options requires an index to be reset (if it already exists)

**Reply by nthonymiller** [reply]
Hi, 

Thanks for response, however this has not worked. Using latest version 5.4 of RavenDb

Receive the following error in the Index Configuration tab - "Unrecognized per-index configuration option. Might be ignored by the server."

The NGram size has not increased.

Note: I deleted the index and recreated. Plus also tried adding Indexing configuration as well. The dropdown shows "Indexing.Analyzers.Lucene.NGram.MaxGram" as the option.

Thanks.

**Comment by garayx** [comment]
Hi @nthonymiller 

It seems like there is a bug that those setting are not applied after they are set. 

I got it working by following those steps:

1. set the configurations & save the index
```
Indexing.Lucene.Analyzers.NGram.MinGram
Indexing.Lucene.Analyzers.NGram.MaxGram
```
![image](https://github.com/ravendb/ravendb/assets/16625387/93ae56d7-8a31-4416-ab72-36c0e396e083)

2. Clone the index 
![image](https://github.com/ravendb/ravendb/assets/16625387/ac1adcd3-64a1-421f-9e8d-99f403bf1a81)

3. New index will be created and the configuration will be applied (MinGram is 3 and MaxGram is 5)
![image](https://github.com/ravendb/ravendb/assets/16625387/f6af6b5b-2adb-4473-8f8d-daebd6916f86)


I have opened a ticket to fix that 
https://issues.hibernatingrhinos.com/issue/RavenDB-21276/
